pyxem/utils/segment_utils.py

```diff
@@ -0,0 +1,311 @@
+
+    sep = np.zeros((np.shape(vdf_temp)[0], np.shape(vdf_temp)[1],
+                    (np.max(labels))), dtype='int32')
+    n, i = 1, 0
+    while (np.max(labels)) > n - 1:
+        sep_temp = labels * (labels == n) / n
+        sep_temp = np.nan_to_num(sep_temp)
+        # Discard a segment if it is too small, or else add it to the list
+        # of separated segments.
+        if ((np.sum(sep_temp, axis=(0, 1)) < min_size)
+                or (max_size is not None
+                    and np.sum(sep_temp, axis=(0, 1)) > max_size)):
+            sep = np.delete(sep, ((n - i) - 1), axis=2)
+            i = i + 1
+        else:
+            sep[:, :, (n - i) - 1] = sep_temp
+        n = n + 1
+    # Put the intensity from the input VDF image into each segmented area.
+    vdf_sep = np.broadcast_to(vdf_temp.T, np.shape(sep.T)) * (sep.T == 1)
+
+    if plot_on: # pragma: no cover
+        # If segments have been discarded, make new labels that do not
+        # include the discarded segments.
+        if np.max(labels) != (np.shape(sep)[2]) and (np.shape(sep)[2] != 0):
+            labels = sep[:, :, 0]
+            for i in range(1, np.shape(sep)[2]):
+                labels = labels + sep[..., i] * (i + 1)
+        # If no separated particles were found, set all elements in
+        # labels to 0.
+        elif np.shape(sep)[2] == 0:
+            labels = np.zeros(np.shape(labels))
+
+        seps_img_sum = np.zeros_like(vdf_temp).astype('float64')
+        for l, vdf in zip(np.arange(1, np.max(labels) + 1), vdf_sep):
+            mask_l = np.zeros_like(labels).astype('bool')
+            mask_l[np.where(labels == l)] = 1
+            seps_img_sum += vdf_temp * mask_l /\
+                np.max(vdf_temp[np.where(labels == l)])
+            seps_img_sum[np.where(labels == l)] += l
+
+        maxi_coord = np.where(local_maxi)
+
+        fig, axes = plt.subplots(2, 3, sharex=True, sharey=True)
+        ax = axes.ravel()
+
+        ax[0].imshow(vdf_temp, cmap=plt.cm.magma_r)
+        ax[0].axis('off')
+        ax[0].set_title('VDF')
+
+        ax[1].imshow(mask, cmap=plt.cm.gray_r)
+        ax[1].axis('off')
+        ax[1].set_title('Mask')
+
+        ax[2].imshow(distance, cmap=plt.cm.gray_r)
+        ax[2].axis('off')
+        ax[2].set_title('Distance and markers')
+        ax[2].imshow(masked_where(markers == 0, markers),
+                     cmap=plt.cm.gist_rainbow)
+        ax[2].plot(maxi_coord1[1], maxi_coord1[0], 'k+')
+        ax[2].plot(maxi_coord[1], maxi_coord[0], 'gx')
+
+        ax[3].imshow(elevation, cmap=plt.cm.magma_r)
+        ax[3].axis('off')
+        ax[3].set_title('Elevation')
+
+        ax[4].imshow(labels, cmap=plt.cm.gnuplot2_r)
+        ax[4].axis('off')
+        ax[4].set_title('Labels')
+
+        ax[5].imshow(seps_img_sum, cmap=plt.cm.magma_r)
+        ax[5].axis('off')
+        ax[5].set_title('Segments')
+
+    return vdf_sep
+
+
+def get_gaussian2d(a, xo, yo, x, y, sigma):
```

Comment:
Couldn't we be reusing code associated with sim_as_signal here?

Reply:
I think `sim_as_signal` got removed when we pulled diffsims out, I wrote a new numpy only implementation but I'm not sure if that would apply here.

Reply:
It would, but I also think this is best left to a later date.